### PR TITLE
[e2e] Malicious dapp warning

### PR DESCRIPTION
**Pull request by BrodyHughes**
Fixes APP-1461

## What changed (plus any additional context for devs)
- Added a check for our malicious dapp
- Was a request of the late Nick (RIP)

## What to test
- Does it pass?

### CODE REVIEW

File: e2e/serial/dappInteractions/5_maliciousDapp.test.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      driver,
+    });
+    expect(button).toBeTruthy();
+    await waitAndClick(button, driver);
+    await delayTime('long');
+
+    // todo: clean this xpath up
+    const rainbowButton = await findElementByXPath(
+      driver,
+      '/html/body/div[2]/div/div/div[2]/div/div/div/div/div[1]/div[2]/div[2]/div[1]/button/div/div',
```
Comment (estebanmino) [inline e2e/serial/dappInteractions/5_maliciousDapp.test.ts:100]: shy xpath and not testid?

Comment (BrodyHughes) [reply e2e/serial/dappInteractions/5_maliciousDapp.test.ts:100]: this is on wayne's test dapp and the test ID wasn't working for me 😅 Let me try again tho bc it's been a few weeks. I believe it was the RK test ID: `rk-wallet-option-rainbow`

File: e2e/serial/dappInteractions/5_maliciousDapp.test.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    await findElementByTestIdAndClick({ id: 'connect-to-hardhat', driver });
+    const button = await findElementByText(driver, 'Disconnect from Hardhat');
+    expect(button).toBeTruthy();
+    await findElementByTestIdAndClick({
+      id: 'navbar-button-with-back',
+      driver,
+    });
+  });
+
+  it('should be able to navigate to the malicious app and click connect', async () => {
```
Comment (DanielSinclair) [inline e2e/serial/dappInteractions/5_maliciousDapp.test.ts:83]: @BrodyHughes Seeing an e2e test failure here on Firefox:
>  × e2e/serial/dappInteractions/5_maliciousDapp.test.ts > App interactions flow > should be able to navigate to the malicious app and click connect
   → Unable to locate window: 
1719991725269	Marionette	INFO	Stopped listening on port 33833
 × e2e/serial/dappInteractions/5_maliciousDapp.test.ts > App interactions flow > should be able to navigate to the malicious app and click connect
   → Unable to locate window:

Comment (BrodyHughes) [reply e2e/serial/dappInteractions/5_maliciousDapp.test.ts:83]: weird. i only tested it on ff locally lol i'll see whats up.

**Comment by DanielSinclair** [comment]
@BrodyHughes Looks like Firefox errors are still popping up here; oddly only on that platform.

**Review by DanielSinclair — APPROVED** [review]
Tests are passing now since we deprecated Firefox; didn't review for quality.

**Review by estebanmino — APPROVED** [review]
